test/acceptance/mixins/search.js

```diff
@@ -20,6 +20,19 @@ module.exports = (target) => {
     this.submitMarriageSearchPage(systemNumber, surname, forenames, dom);
   };
 
+  target.marriageSearchWithRoleHeader = function(roles) {
+    this.execute((u, r) => {
+      /* eslint-env browser */
+      var client = new XMLHttpRequest();
+      client.open('GET', u, false);
+      client.setRequestHeader('X-Auth-Roles', r);
+      client.send();
+      document.body.parentElement.innerHTML = client.responseText;
+      /* eslint-env node, mocha */
+    }, this.getUrl(), roles);
+    this.waitForExist('table.details', 5000);
+  };
```

Comment:
I think we need to communicate in the code that something odd is going on.

Maybe something like `jsRefreshWithRoles()` calls `jsRefreshWithHeaders()` and have the test navigate to the page under test without the role and then call the magical code to get a version with the role in place.

That way it will be clearer to future LEVOps that something odd is going on to make the test work.

Reply:
That's the way the tests work currently. I can look at making the code more explanatory though...

Reply:
It doesn't look like this code has been updated. it probably just needs to be rename (to `refreshWithRole()`?), moved to a non-marriage specific place and changed to wait for a more generic element. (If I understand correctly.)

Reply:
Refactored.